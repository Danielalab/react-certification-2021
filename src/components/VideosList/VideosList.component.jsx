import React from 'react';
import styled from 'styled-components';
import defaultChannelAvatar from '../../assets/images/undraw_default_avatar.svg';

const VideoCard = styled.div`
  max-width: 320px;
  margin: 0.75rem;
`;

const Wrapper = styled.div`
  display: flex;
  align-items: flex-start;
`;

const Link = styled.a`
  color: #000000;
  font-weight: 700;
`;

const ChannelAvatar = styled.img`
  width: 40px;
  height: 40px;
  border-radius: 50%;
`;

const Text = styled.p`
  margin: 0;
  color: #555555;
`;

const MarginX = styled.div`
  margin: 0 0.75rem;
`;

const Video = ({ id, videoData }) => {
  return (
    <VideoCard>
      <figure>
        <img src={videoData.thumbnails.medium.url} alt={`${videoData.title} thumbnail`} />
      </figure>
      <Wrapper>
        <ChannelAvatar src={defaultChannelAvatar} alt={videoData.channelTitle} />
        <MarginX className="margin-x">
          <Link href={`https://www.youtube.com/watch?v=${id}`}>{videoData.title}</Link>
          <div>
            <Text>{videoData.channelTitle}</Text>
            <Text>{videoData.publishedAt}</Text>
          </div>
        </MarginX>
      </Wrapper>
    </VideoCard>
  );
};

const VideosList = ({ videos }) =>
  videos.map(({ id, snippet }) => (
    <Video key={id.videoId} id={id.videoId} videoData={snippet} />
  ));

export default VideosList;
