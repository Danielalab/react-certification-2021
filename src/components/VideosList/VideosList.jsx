import React from 'react';
import styled from 'styled-components';
import { items } from './youtube-videos-mock.json';

const { snippet: channel } = items.shift();

const VideoCard = styled.div`
  max-width: 320px;
  margin: 0.75rem;
  p {
    margin: 0;
    color: #555555;
  }
  .channel-avatar {
    width: 40px;
    height: 40px;
    border-radius: 50%;
  }
  .margin-x {
    margin: 0 0.75rem;
  }
`;

const Wrapper = styled.div`
  display: flex;
  align-items: flex-start;
`;

const Link = styled.a`
  color: #000000;
  font-weight: 700;
`;

const VideosList = () =>
  items.map(({ id, snippet }) => (
    <VideoCard key={id.videoId}>
      <figure>
        <img src={snippet.thumbnails.medium.url} alt={`${snippet.title} thumbnail`} />
      </figure>
      <Wrapper>
        <img
          src={channel.thumbnails.default.url}
          alt={channel.channelTitle}
          className="channel-avatar"
        />
        <div className="margin-x">
          <Link href={`https://www.youtube.com/watch?v=${id.videoId}`}>
            {snippet.title}
          </Link>
          <div>
            <p>{snippet.channelTitle}</p>
            <p>{snippet.publishedAt}</p>
          </div>
        </div>
      </Wrapper>
    </VideoCard>
  ));

export default VideosList;
