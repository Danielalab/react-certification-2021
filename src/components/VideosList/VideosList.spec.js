import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import VideosList from './VideosList.component';
import dataDummy from '../../hooks/tests/responses/searchQuerySuccess.json';

describe('VideosList', () => {
  let container;

  beforeEach(() => {
    container = render(<VideosList videos={dataDummy.items} />, {
      wrapper: MemoryRouter,
    }).container;
  });

  test('Should renders VideosList component', () => {
    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test('Should contains an image with the thumbnail url', () => {
    const videoThumbnailImageElement = screen.getByAltText(
      'fake video title 1 thumbnail'
    );
    expect(videoThumbnailImageElement).toBeInTheDocument();
    expect(videoThumbnailImageElement).toHaveAttribute('src', 'fake-video-url-image.png');
  });

  test('Should contains an image with the channel avatar', () => {
    const channelImageElement = screen.getAllByAltText('fake channel title')[0];
    expect(channelImageElement).toBeInTheDocument();
    expect(channelImageElement).toHaveAttribute('src', 'undraw_default_avatar.svg');
  });

  test('Should contains a link redirects to the video url', () => {
    const linkELement = screen.getByText(/fake video title 1/i);
    expect(linkELement).toBeInTheDocument();
    expect(linkELement).toHaveAttribute('href', '/fakeVideoId1');
  });

  test('Should shows the channel title', () => {
    const channelTitleElement = screen.getAllByText(/fake channel title/i)[0];
    expect(channelTitleElement).toBeInTheDocument();
  });

  test('Should shows the video published date', () => {
    const channelTitleElement = screen.getAllByText(/2014-09-27T01:39:18Z/i)[0];
    expect(channelTitleElement).toBeInTheDocument();
  });
});
