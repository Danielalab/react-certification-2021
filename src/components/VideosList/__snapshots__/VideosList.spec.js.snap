// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideosList Should renders VideosList component 1`] = `
<div>
  <div
    class="sc-bdnxRM iIRjbH"
  >
    <figure>
      <img
        alt="fake video title 1 thumbnail"
        src="fake-video-url-image.png"
      />
    </figure>
    <div
      class="sc-gtsrHT eMBaCB"
    >
      <img
        alt="fake channel title"
        class="sc-hKFxyN bdMQIT"
        src="undraw_default_avatar.svg"
      />
      <div
        class="sc-jSFjdj bgycdW margin-x"
      >
        <a
          class="sc-dlnjwi hzuETT"
          href="/fakeVideoId1"
        >
          fake video title 1
        </a>
        <div>
          <p
            class="sc-eCApnc fpRBdY"
          >
            fake channel title
          </p>
          <p
            class="sc-eCApnc fpRBdY"
          >
            2014-09-27T01:39:18Z
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="sc-bdnxRM iIRjbH"
  >
    <figure>
      <img
        alt="fake video title 2 thumbnail"
        src="fake-video-url-image.png"
      />
    </figure>
    <div
      class="sc-gtsrHT eMBaCB"
    >
      <img
        alt="fake channel title"
        class="sc-hKFxyN bdMQIT"
        src="undraw_default_avatar.svg"
      />
      <div
        class="sc-jSFjdj bgycdW margin-x"
      >
        <a
          class="sc-dlnjwi hzuETT"
          href="/fakeVideoId2"
        >
          fake video title 2
        </a>
        <div>
          <p
            class="sc-eCApnc fpRBdY"
          >
            fake channel title
          </p>
          <p
            class="sc-eCApnc fpRBdY"
          >
            2014-09-27T01:39:18Z
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
